web/src/stream_edit.js

```diff
@@ -39,8 +39,20 @@ import * as user_groups from "./user_groups";
 import {user_settings} from "./user_settings";
 import * as util from "./util";
 
-function setup_subscriptions_stream_hash(sub) {
-    const hash = hash_util.stream_edit_url(sub);
+export const right_section_url_fragment_to_section_name = new Map([
+    ["personal", "personal_settings"],
+    ["subscribers", "subscriber_settings"],
+    ["general", "general_settings"],
+]);
+
+export const right_section_name_to_url_fragment = new Map([
+    ["personal_settings", "personal"],
+    ["subscriber_settings", "subscribers"],
+    ["general_settings", "general"],
+]);
```

Comment:
It would be cool to have a shared data structure instead of writing out the pairs twice like this, but I couldn't think of something nice to do.

Reply:
I think the better plan would be to just make the URL fragment names be the same as the code-level tab names.

Reply:
Not sure which direction to take this. Asked on CZO [here](https://chat.zulip.org/#narrow/stream/9-issues/topic/stream.20settings.20url/near/1681645)

Reply:
Decided to use the shorter strings (e.g. `general`) for both.